components/de_amp/browser/de_amp_body_handler.cc

```diff
@@ -128,7 +129,7 @@ DeAmpBodyHandler::Action DeAmpBodyHandler::OnBodyUpdated(
 
   switch (state_) {
     case State::kCheckForAmp:
-      if (!CheckIfAmpPage(body)) {
+      if (!CheckIfAmpPage(std::string_view(body).substr(0, kMaxBytesToCheck))) {
```

Comment:
Data usually comes here at 64 kb, but this is not guaranteed, so there may be more data than usual at the time of verification. For example, if we received 32 kb 5 times and then 64 kb, there will be more than 196 kb in the buffer.